posts/2023/installing-farmos-on-ddev-with-composer/index.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Installing farmOS on DDEV with Composer
+date: 2023-05-17
+author: rgcarr
+slug: 2023/installing-farmos-on-ddev-with-composer
+---
+
+# DDEV
+
+For the recommended farmOS local development environment using Docker and Docker compose, see the 
+[farmOS development environment docs](https://farmos.org/development/environment). That method works great in
+supported environments for developing farmOS core and small projects but can
+require additional configuration when working on larger projects in different
+environments like Colima. For these use-cases installing farmOS in a
+[DDEV](https://ddev.com/) environment may be more straightforward.
+
+> DDEV is an open source tool for running local web development environments for PHP, Python and Node.js, ready in minutes. Its powerful, flexible per-project environment configurations can be extended, version controlled, and shared. DDEV allows development teams to adopt a consistent Docker workflow without the complexities of bespoke configuration.
+
+DDEV's many features are what make it a great way to build Drupal and farmOS
+projects in a local environment:
+
+* Cross-platform, open source software.
+* Supports many versions of Docker or can be used with Colima.
+* When combined with Colima and Mutagen, performance is significantly improved.
+* Automatic HTTPS.
+* Run multiple projects concurrently.
+* Can be used to develop many other CMSs.
+* Support multiple databases (including MySQL, Postgre, SQLite).
+* Lots of developer tools included, and is easy to extend and integrate.  
+
+# Installing farmOS with DDEV and Composer
+
+- First, follow the instructions to install and setup DDEV: https://ddev.readthedocs.io/en/latest/users/install/
+- Once installed, the following commands will create a DDEV project and install farmOS:
+
+```
+# Create a directory for the farmOS DDEV project
+mkdir -p farmos-project && cd farmos-project
```

Comment:
When creating just a single directory right in the current working directory, the `-p` argument is not needed.
```suggestion
mkdir farmos-project && cd farmos-project
```